include/jwt-cpp/jwt.h

```diff
@@ -90,19 +90,26 @@ namespace jwt {
 	 * \brief Everything related to error codes issued by the library
 	 */
 	namespace error {
-		struct signature_verification_exception : public std::system_error {
+		/**
+		 * \brief Generic base class for all JWT specific exceptions
+		 */
+		struct exception : public std::exception {
+			using std::exception::exception;
+		};
+
+		struct signature_verification_exception : exception, public std::system_error {
```

Comment:
This feels really weird cause now signature_verification_exception inherits from std::exception twice.
Could we get away with inheriting `exception` from `std::system_error` (which inherits `std::runtime_error`, which in turn inherits `std::exception`) ?